packages/opencensus-nodejs/test/test-tracing.ts

```diff
@@ -0,0 +1,203 @@
+      assert.strictEqual(defaultConfig.logLevel, tracing.config.logLevel);
+      assert.strictEqual(
+          defaultConfig.maximumLabelValueSize,
+          tracing.config.maximumLabelValueSize);
+      assert.strictEqual(
+          defaultConfig.samplingRate, tracing.config.samplingRate);
+      assert.ok(tracing.config.plugins['http']);
+      assert.strictEqual(
+          tracing.config.plugins['http'],
+          `${Constants.OPENCENSUS_SCOPE}/${
+              Constants.DEFAULT_PLUGIN_PACKAGE_NAME_PREFIX}-http`);
+
+      assert.ok(tracing.config.plugins['https']);
+      assert.strictEqual(
+          tracing.config.plugins['https'],
+          `${Constants.OPENCENSUS_SCOPE}/${
+              Constants.DEFAULT_PLUGIN_PACKAGE_NAME_PREFIX}-https`);
+    });
+
+    it('should start tracing with a non-default logLevel', () => {
+      aTracing = tracing.start({logLevel: 3});
+      assert.strictEqual(tracing.config.logLevel, 3);
+      const consoleLogger = aTracing.tracer.logger as logger.ConsoleLogger;
+      assert.strictEqual(consoleLogger.level, 'info');
+    });
+
+    it('should start tracing with a logger instance', () => {
+      const aLogger = logger.logger('debug');
+      aTracing = tracing.start({logger: aLogger});
+      assert.strictEqual(tracing.config.logger, aLogger);
+      const consoleLogger = aTracing.tracer.logger as logger.ConsoleLogger;
+      assert.strictEqual(consoleLogger.level, 'debug');
+    });
+
+    it('should start with an exporter instance', () => {
+      aTracing = tracing.start({exporter: NOOP_EXPORTER});
+      assert.strictEqual(tracing.config.exporter, NOOP_EXPORTER);
+      assert.strictEqual(aTracing.exporter, NOOP_EXPORTER);
+    });
+
+    it('should start with a non-default bufferSize', () => {
+      const bufferSizeValue = defaultConfig.bufferSize + 1;
+      tracing.start({bufferSize: bufferSizeValue});
+      assert.strictEqual(tracing.config.bufferSize, bufferSizeValue);
+    });
+
+    it('should start with a non-default bufferTimeout', () => {
+      const bufferTimeoutValue = defaultConfig.bufferTimeout + 100;
+      tracing.start({bufferTimeout: bufferTimeoutValue});
+      assert.strictEqual(tracing.config.bufferTimeout, bufferTimeoutValue);
+    });
+
+    it('should start with a non-default maximumLabelValueSize', () => {
+      const maximumLabelValueSizeValue =
+          defaultConfig.maximumLabelValueSize + 10;
+      tracing.start({maximumLabelValueSize: maximumLabelValueSizeValue});
+      assert.strictEqual(
+          tracing.config.maximumLabelValueSize, maximumLabelValueSizeValue);
+    });
+
+    it('should start with a non-default samplingRate', () => {
+      const samplingRateValue = defaultConfig.samplingRate / 100;
+      tracing.start({samplingRate: samplingRateValue});
+      assert.strictEqual(tracing.config.samplingRate, samplingRateValue);
+    });
+
+    it('should start with an end-user plugin list', () => {
```

Comment:
Can you change `end-user` to `user-provided` for more clarity?

Reply:
Done!